### PR TITLE
Allow people importing cortex to add their own chunk formats.

### PR DESCRIPTION
**Pull request by tomwilkie**
Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

**Comment by bboreham** [comment]
How do we ensure that the map is not written at the same time as being read?

**Comment by tomwilkie** [comment]
I've so far only used this in `init()` functions.  Will add to comment.

File: pkg/chunk/encoding/factory.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	if !ok {
+		return nil, fmt.Errorf("unknown chunk encoding: %v", encoding)
+	}
+
+	return enc.New(), nil
+}
+
+// MustRegisterEncoding add a new chunk encoding.  There is no locking, so this
+// must be called in init().
+func MustRegisterEncoding(enc Encoding, name string, new func() Chunk) {
```
Comment (gouthamve) [inline pkg/chunk/encoding/factory.go]: `new` is also a keyword. Can we name this `newFunc` or something similar to avoid confusion?

Comment (tomwilkie) [reply pkg/chunk/encoding/factory.go]: Fair enough - although I tend to dislike names which include types (`errChan`, `newFunc` etc).  I've called it `f`.

**Pull request merged**